src/Microsoft.AspNetCore.SignalR.Core/Internal/SyntheticArgumentsManager.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+
+namespace Microsoft.AspNetCore.SignalR.Internal
+{
+    internal class SyntheticArgumentsManager
```

Comment:
This didn't turn out as nice as I was hoping. Let's chat a bit later today and see if we can find a clean pattern here.

Reply:
Remove this class.